### PR TITLE
[TIMOB-25683] iOS: Refactor Ti.Filesystem, fix broken tests, use ARC

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25683

### CODE REVIEW

File: iphone/Classes/TiFile.m [hunk 1]
```diff
@@ -63,7 +61,7 @@ - (unsigned long long)size
 
 - (id)blob
 {
-  return [[[TiBlob alloc] _initWithPageContext:[self pageContext] andFile:path] autorelease];
+  return [[TiBlob alloc] _initWithPageContext:[self pageContext] andFile:_path];
```
Comment (vijaysingh-axway) [inline iphone/Classes/TiFile.m]: Why autorelease removed? will it not cause memory leak?

Comment (hansemannn) [reply iphone/Classes/TiFile.m]: I moved the whole class to ARC

Comment (vijaysingh-axway) [reply iphone/Classes/TiFile.m]: Cool. Can you once check your test case in ticket. It is giving error for testRecursiveDirectories() and for others its giving 0 but you have mentioned it should be Double. Am I missing anything? Thanks!

Comment (hansemannn) [reply iphone/Classes/TiFile.m]: I think there are some issues with the change in general, some paths are released too early. Will put into progress again for now.

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]
As per comment in PR, there is some issue.

**Comment by hansemannn** [comment]
Closing in favor of #10019

**Pull request closed without merging**